File: user-service/src/main/java/rs/edu/raf/model/user/BankAccountHolder.java
package rs.edu.raf.model.user;

import jakarta.persistence.*;
import lombok.Getter;
import lombok.Setter;

@Entity
@Inheritance(strategy = InheritanceType.JOINED)
@Getter
@Setter
public abstract class BankAccountHolder {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    @Column(unique = true)
    private String phoneNumber;
    private String address;
}

File: user-service-kotlin/src/main/kotlin/org/example/userservicekotlin/model/user/BankAccountHolder.kt
package org.example.userservicekotlin.model.user

import jakarta.persistence.*

@Entity
@Inheritance(strategy = InheritanceType.JOINED)
open class BankAccountHolder (
    @Id @GeneratedValue(strategy = GenerationType.IDENTITY) val id: Long?,
    @Column(unique = true) var phoneNumber: String,
    var address: String
) {
}